### DISCUSSION TITLE
Marketing the project as open source, while not using an open source license

### DISCUSSION THREAD

**Discussion by ssddanbrown** [Helicone/helicone#165] [General]
Hi 👋,

I noticed the project, advertising itself as open source. As seen on [your hacker news post](https://news.ycombinator.com/item?id=35279155), [within your readme](https://github.com/Helicone/helicone/blob/f443b201c422952be2de6aa30f2c1a08aacbf88f/README.md?plain=1#L10) and on your website. 

Just wanted to query this usage since your project [uses the commons clause](https://github.com/Helicone/helicone/blob/f443b201c422952be2de6aa30f2c1a08aacbf88f/LICENSE#L5) and hence puts limitations on usage by preventing the ability to sell the software. This goes against the commonly understood [open source definition](https://opensource.org/osd/) (Which generally ensures open use, distribution & modification).

The [commons clause site](https://commonsclause.com/) even explicitly states that the license is not open source:

> Is this “Open Source”?
> No.
> “Open source”, has a [specific definition](https://opensource.org/osd) that was written years ago and is stewarded by the Open Source Initiative, which approves Open Source licenses. Applying the Commons Clause to an open source project will mean the source code is available, and meets many of the elements of the Open Source Definition, such as free access to source code, freedom to modify, and freedom to re-distribute, but not all of them. So to avoid confusion, it is best not to call Commons Clause software “open source.”

Just to be clear, I have no issue with folks setting restrictions/limitations in the interest of ensuring sustainable work to protect their business where required. It's good to be somewhat open, even if not fully. It's just a shame to see the term open source stretched to non-open-source scenarios. If it helps, I've [written more about this here](https://danb.me/blog/posts/why-open-source-term-is-important/). Such projects are often instead referred to as "source available" or "fair code".

I'm just raising here to make you aware of this potentially confusing use of open source, or are you already aware of this and intentionally choosing to use "open source" while using the commons clause license?

**Comment by chitalian** [comment]
Hi @ssddanbrown. Thank you so much for flagging this! 

We are still having an internal discussion about the right strategy here, but in the meanwhile, we open-sourced everything except for the contents of `/web`.

We plan to open source more of the `/web` directory but I am curious about your thoughts on having a large portion of the codebase open-source, but not everything and still marketing as open-source.

**Reply by ssddanbrown** [reply]
Hi @chitalian,
Thanks for the response!

> ... we open-sourced everything except for the contents of /web.

Awesome! But right now this is really unclear without users having to do digging into the files. GitHub will pick up the top-level license and denotes the repo as MIT. Ideally you should mention this within the top-level `LICENSE` file since that's where folks will actually look.

> I am curious about your thoughts on having a large portion of the codebase open-source, but not everything and still marketing as open-source.

Are you advertising your application (Including the web portion) as a whole as open source? If so, I'd consider that to still be misleading. I'd expect to be able to take and use the application as open source as a whole. I'm not exactly sure on the value and requirement of various components though.

Based on the question asked I get the feeling you want to use "open source" for marketing while adding limitations to specifically protect yourselves and give yourselves more rights over distribution/modification/use, but these are essentially conflicting desires.

**Reply by chitalian** [reply]
Hi @ssddanbrown Thank you so much for getting back to us!

We ended up just keeping things simple and going fully OSS
https://github.com/Helicone/helicone/pull/227

Thanks so much for your advice, feedback and help.
